internal/pkg/cli/deploy.go

```diff
@@ -4,26 +4,180 @@
 package cli
 
 import (
+	"fmt"
+	"strings"
+
 	"github.com/aws/copilot-cli/cmd/copilot/template"
+	"github.com/aws/copilot-cli/internal/pkg/aws/sessions"
 	"github.com/aws/copilot-cli/internal/pkg/cli/group"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+	"github.com/aws/copilot-cli/internal/pkg/term/command"
+	termprogress "github.com/aws/copilot-cli/internal/pkg/term/progress"
+	"github.com/aws/copilot-cli/internal/pkg/term/prompt"
+	"github.com/aws/copilot-cli/internal/pkg/term/selector"
+	"github.com/aws/copilot-cli/internal/pkg/workspace"
 	"github.com/spf13/cobra"
 )
 
-// BuildDeployCmd is the deploy command - which is
-// an alias for app deploy.
-func BuildDeployCmd() *cobra.Command {
-	deployCmd := buildSvcDeployCmd()
-	deployCmd.Use = "deploy"
-	deployCmd.Short = "Deploy your service."
-	deployCmd.Long = `Command for deploying services to your environments.`
-	deployCmd.Example = `
-	Deploys a service named "frontend" to a "test" environment.
-	/code $ copilot deploy --name frontend --env test`
+const (
+	svcWkldType = "svc"
+	jobWkldType = "job"
+)
+
+type deployOpts struct {
+	deployWkldVars
+
+	deployWkld     actionCommand
+	setupDeployCmd func(*deployOpts, bool)
+
+	sel    wsSelector
+	store  store
+	ws     wsWlDirReader
+	prompt prompter
+
+	// values for logging
+	wlType string
+}
+
+func newDeployOpts(vars deployWkldVars) (*deployOpts, error) {
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("new config store: %w", err)
+	}
 
-	deployCmd.SetUsageTemplate(template.Usage)
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, fmt.Errorf("new workspace: %w", err)
+	}
+	prompter := prompt.New()
+	if err != nil {
+		return nil, err
+	}
+	return &deployOpts{
+		deployWkldVars: vars,
+		store:          store,
+		sel:            selector.NewWorkspaceSelect(prompter, store, ws),
+		ws:             ws,
+		prompt:         prompter,
+
+		setupDeployCmd: func(o *deployOpts, isJob bool) {
```

Comment:
I'm trying to think how we can make this scale better if in the future we add a new type. Maybe this?
```go
setupDeployCmd: func(o *deployOpts, workloadType string) {
   switch t := wlType; {
   case contains(manifest.JobTypes, t):
       ...
   case contains(manifest.ServiceTypes, t):
       ...
   }
}
```